### ISSUE TITLE
use_build_context_synchronously false positive when using switch

### ISSUE THREAD

**Issue by quoc-huynh-cosee** [dart-lang/linter#4794]
**Describe the issue**
I'm receiving a `use_build_context_synchronously` lint warning using `switch` with the given example, although there is no async gap.

**To Reproduce**
```dart
Future<void> doSomething() async {}

enum State { one, two, three }

Future<void> myMethod(BuildContext context, State state) async {
  switch (state) {
    case State.one:
      final size = MediaQuery.of(context).size; // No warning
    case State.two:
      await doSomething();
    case State.three:
      final size = MediaQuery.of(context).size; // Lint warning
  }
}
```
**Expected behavior**
No lint warning for `case State.three`.

**Comment by srawlins** [comment]
@quoc-huynh-cosee thanks for the report, can you say what version of Flutter or Dart you are using?

**Comment by srawlins** [comment]
Thank you! This was fixed with https://github.com/dart-lang/linter/pull/4547 and should be available in Dart 3.2.0.
